### ISSUE TITLE
Make type for initial value an additional template argument

### ISSUE THREAD

**Issue by jdgarciauc3m** [arcosuc3m/grppi#199]
Instead of using meta-programming tricks, just add an additional template parameter for the initial type

Example:
~~~c++
template < typename InputIt, typename Combiner, typename IdentityType>
auto reduce(sequential_execution & ex, InputIt first, InputIt last, 
            IdentityType identity,
            Combiner && combine_op)
{
~~~

**Comment by Drioastorga** [comment]
This was solved by a previous pull request.

**Comment by jdgarciauc3m** [comment]
Solved by PR #203
